### ISSUE TITLE
[Bug]: Compaction cannot be done due to statslog number check

### ISSUE THREAD

**Issue by congqixia** [milvus-io/milvus#24834]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: bc95cab
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

All compaction will fail due to binlog/statslog number check reporting 
```
segment can not be saved because of binlog file not match stat log number
```

### Expected Behavior

Compaction shall be doable with merged statslog

**Comment by xiaofan-luan** [comment]
should we also backport this to 2.2?

**Comment by congqixia** [comment]
> should we also backport this to 2.2?

Double checked with @aoiasd, merged statslog was not in 2.2 branch
